From 5789bb35b08bb126e12dae67506c9b1cc24cb99e Mon Sep 17 00:00:00 2001
From: Cedric Poignet <cedric.poignet@intel.com>
Date: Mon, 19 Aug 2013 14:27:11 +0200
Subject: crashtool: log bluetooth fw version (libbt)

BZ: 108245

Logging bluetooth fw version.

Category: aosp improvement
Domain: BT-Core
Origin: internal
Upstream-Candidate: no, not-aosp

Change-Id: If165312dae9baa4950aab10892a0926e3c183f2b
Orig-Change-Id: I0697f13d44b10b9206e41d7d2bfa79993598b676
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/126132
Reviewed-by: Poignet, Cedric <cedric.poignet@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 src/hardware.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/hardware.c b/src/hardware.c
index a4496be..d57ac0b 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -835,6 +835,7 @@ void hw_config_cback(void *p_mem)
                 {
                     uint16_t    lmp_subversion;
                     uint8_t     *p_lmp;
+                    char tmp[5];
 
                     p_lmp = (uint8_t *) (p_evt_buf + 1) + HCI_EVT_CMD_CMPL_LOCAL_REVISION;
                     STREAM_TO_UINT16(lmp_subversion, p_lmp);
@@ -844,6 +845,9 @@ void hw_config_cback(void *p_mem)
                         /* Found BCM4335B0 revision */
                         hw_cfg_cb.local_chip_name[7] = 'B';
                     }
+
+                    snprintf(tmp, sizeof(tmp), "%04x", lmp_subversion);
+                    lct_log(CT_EV_INFO, "cws.bt", "fw_version", 0, hw_cfg_cb.local_chip_name, tmp);
                 }
                 /* fall through intentionally */
 
-- 
1.7.4.1

